vcf-taggable-text-area/src/main/java/org/vaadin/addons/componentfactory/tta/FilterListSelector.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * This particular selector is advised to be used when a large amount of items needs to be displayed.
+ * 
+ * @param <T> the type of items displayed 
+ */
+@SuppressWarnings("serial")
+@CssImport(value = "./styles/filter-list-selector.css", themeFor = "vaadin-grid")
+public class FilterListSelector<T> extends BaseFilterListSelector<T> {
+	
+	private Grid<T> gridList = new Grid<>();
+	private Column<T> column;
+    private ComponentRenderer<? extends Component, T> itemRenderer;
+	private T selectedItem = null;
+	private T focusedItem = null;
+	private int visibleItems = 10;
+	
+	/**
+     * Constructs a {@code FilterListSelector} with the given list of items.
+     *
+     * @param items the list of items to display in the {@link Grid}
+     */
+    public FilterListSelector(List<T> items) {
+        this(items, new ComponentRenderer<>(item -> {
+          HorizontalLayout row = new HorizontalLayout();
+          row.setAlignItems(FlexComponent.Alignment.CENTER);
+
+          Span name = new Span(""+item);
+
+          VerticalLayout column = new VerticalLayout(name);
+          column.setPadding(false);
+          column.setSpacing(false);
+
+          row.add(column);
+          row.getStyle().set("line-height", "var(--lumo-line-height-m)");
+          return row;
+      }));
+    }
+    
+    /**
+     * Constructs a {@code FilterListSelector} with the given list of items and the given itemRenderer.
+     *
+     * @param items the list of items to display in the {@link Grid}
+     * @param itemRenderer the renderer for the items
+     */
+	public FilterListSelector(List<T> items, ComponentRenderer<? extends Component, T> itemRenderer) {
+	    setClassName("taggable-textarea-filter-list-selector");
+        this.itemRenderer = itemRenderer;
+                
+        gridList.addClassName("taggable-textarea-filter-list-selector-list");
+        gridList.setSelectionMode(SelectionMode.SINGLE);
+        gridList.setItems(items);
+        gridList.addThemeVariants(GridVariant.LUMO_NO_ROW_BORDERS, GridVariant.LUMO_NO_BORDER);
+        gridList.setAllRowsVisible(items.size() < visibleItems);
+        column = gridList.addColumn(this.itemRenderer);
+        column.setClassNameGenerator(c -> "taggable-textarea-filter-list-selector-list-column");
+
+        initFilter(items);
+		filter.addValueChangeListener(e -> {
+			filteredItems = items.stream()
+					.filter(item -> getFilterExpression().apply(item, filter.getValue()))
+					.collect(Collectors.toList());
+			gridList.setItems(filteredItems);
+			gridList.setAllRowsVisible(filteredItems.size() < visibleItems);
+		});
+		filter.getElement().executeJs(
+		    "this.addEventListener('keydown', (event) => {" +
+		    "    if (event.key === 'ArrowDown') { " +
+		    "        const grid = $0;" +
+		    "        requestAnimationFrame(() => { " +
+		    "            const firstCell = grid.shadowRoot.querySelector('[class=\"taggable-textarea-filter-list-selector-list-column\"]');" +
```

Comment:
The selector can be `.taggable-textarea-filter-list-selector-list-column`

Reply:
Good catch. Fixed.